concent_api/core/admin.py

```diff
@@ -1,6 +1,119 @@
 from django.contrib import admin
 
+from .models import PendingResponse
 from .models import StoredMessage
+from .models import Subtask
 
 
+class ReadOnlyMixin(object):
+    """ Disables all editing capabilities. """
+
+    def get_actions(self, request):
+        actions = super().get_actions(request)
+        if 'delete_selected' in actions:
+            del actions['delete_selected']
+        return actions
+
+    def has_add_permission(self, _request, _obj = None):  # pylint: disable=no-self-use
+        return False
+
+    def has_change_permission(self, request, _obj=None):  # pylint: disable=no-self-use
+        return request.method != 'POST'
+
+    def has_delete_permission(self, _request, _obj = None):  # pylint: disable=no-self-use
+        return False
+
+    def save_model(self, request, obj, form, change):  # pylint: disable=no-self-use
+        pass
+
+    def delete_model(self, request, obj):  # pylint: disable=no-self-use
+        pass
+
+    def save_related(self, request, form, formsets, change):  # pylint: disable=no-self-use
+        pass
+
+
+class ActivePassiveStateFilter(admin.SimpleListFilter):
+
+    title          = 'Active/passive state'
+    parameter_name = 'active_passive_state'
+
+    def lookups(self, request, model_admin):
+        return (
+            ('active',  'Active'),
+            ('passive', 'Passive'),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == 'active':
+            return queryset.filter(state__in = Subtask.ACTIVE_STATES)
+        elif self.value() == 'passive':
+            return queryset.filter(state__in = Subtask.PASSIVE_STATES)
+        return queryset
+
+
+class SubtaskAdmin(ReadOnlyMixin, admin.ModelAdmin):
+
+    list_display = [
+        'subtask_id',
+        'task_id',
+        'state',
+        'get_provider_public_key',
+        'get_requestor_public_key',
+        'next_deadline',
+    ]
+    list_filter = (
+        ActivePassiveStateFilter,
+        'state',
+    )
+    search_fields = [
+        'provider__public_key',
+        'requestor__public_key',
+        'subtask_id',
+        'task_id',
+    ]
+
+    def get_provider_public_key(self, obj):  # pylint: disable=no-self-use
+        return obj.provider.public_key
+    get_provider_public_key.short_description = 'Provider public key'  # type: ignore
+
+    def get_requestor_public_key(self, obj):  # pylint: disable=no-self-use
+        return obj.provider.public_key
+    get_requestor_public_key.short_description = 'Requestor public key'  # type: ignore
+
+
+class PendingResponseAdmin(ReadOnlyMixin, admin.ModelAdmin):
+
+    list_display = [
+        'response_type',
+        'queue',
+        'get_subtask_subtask_id',
+        'get_client_public_key',
```

Comment:
It is not working like this. It will not create link automatically. When you define list items like this, django admin evaluates those models `__str__` method, which represents the model. Currently we don't have those, and even if we want to have I don't think `Subtask` model should be represented only by `subtask_id`. I have used those because I wanted to show explicitly those values there.

Reply:
We definitely should have `__str__()` methods and they should print more than just `subtask_id`. It worked quite nicely in previous projects that way in the admin. But OK, we probably don't want it to include the key. Let's leave it as is in that case.